modin/core/dataframe/algebra/binary.py

```diff
@@ -94,11 +94,21 @@ def caller(
                         )
                     )
                 else:
+                    if other.dtypes is not None and query_compiler.dtypes is not None:
+                        if func.__name__ == "add":
+                            dtypes = pandas.Series(
+                                map(
+                                    lambda x: np.find_common_type([*x], []),
+                                    zip(other.dtypes, query_compiler.dtypes),
+                                )
+                            )
```

Comment:
1. Which binary operations do not follow this logic for computing resultant dtypes?
2. Why should we use `np.find_common_type` instead of `pandas.core.dtypes.cast.find_common_type`?

Reply:
1)Binary operations such as divide(returns a float even if both operands are int), modulus etc might not follow this logic. I will collate a list of operators.
2)Tried to use pandas.core.dtypes.cast.find_common_type directly on dtypes of both series but this was not working in 1.5.2 version of pandas 

```
pd.core.dtypes.cast.find_common_type([df1.dtypes,df3.dtypes]) #error in pandas 1.5.2
pd.core.dtypes.cast.find_common_type([df1.dtypes,df3.dtypes]) #works for pandas 1.3.x
```
So decided to go with np.find_common_type as dtypes were used one at a time, I will change this to pd.core.dtypes.cast.find_common_type if this is the preffered way.

Reply:
Surprising to me most binary operators even including modulus and floordivision seems to follow the pd.core.dtypes.cast.find_common_type so i have included  these too in the PR. Below are few binary operations which are different.
1)division
2)boolean operators such as equals not equals, > , >=, < ,<= etc

Reply:
What scenarios is `find_common_type` failing in, and what error message do you get? Modin uses this method internally in a few places, so I'm a little surprised there aren't existing bugs for it.

Reply:
I could see `find_common_type` failing  when  passing 2 series objects as arguments (as mentioned in my previous post). The current code with zip works for pd.core.dtypes.cast.find_common_type as well.

> 1)Binary operations such as divide(returns a float even if both operands are int), modulus etc might not follow this logic. I will collate a list of operators. 2)Tried to use pandas.core.dtypes.cast.find_common_type directly on dtypes of both operand dataframes but this was not working in 1.5.2 version of pandas
> 
> ```
> pd.core.dtypes.cast.find_common_type([df1.dtypes,df3.dtypes]) #error in pandas 1.5.2
> pd.core.dtypes.cast.find_common_type([df1.dtypes,df3.dtypes]) #works for pandas 1.3.x
> ```
> 
> So decided to go with np.find_common_type as dtypes were anyways used one at a time, I can change this to pd.core.dtypes.cast.find_common_type if this is the preffered way.




This is the error message.

**ValueError: The truth value of a Series is ambiguous. Use a.empty, a.bool(), a.item(), a.any() or a.all().**

### Full traceback.

_---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-3-a67402f7da40> in <module>
----> 1 pd.core.dtypes.cast.find_common_type([df1.dtypes,df3.dtypes])
      2 #pd.core.dtypes.cast.find_common_type([other.dtypes, query_compiler.dtypes])

2 frames
/usr/local/lib/python3.8/dist-packages/pandas/core/dtypes/cast.py in find_common_type(types)
   1611     # workaround for find_common_type([np.dtype('datetime64[ns]')] * 2)
   1612     # => object
-> 1613     if lib.dtypes_all_equal(list(types)):
   1614         return first
   1615 

/usr/local/lib/python3.8/dist-packages/pandas/_libs/lib.pyx in pandas._libs.lib.dtypes_all_equal()

/usr/local/lib/python3.8/dist-packages/pandas/core/generic.py in __nonzero__(self)
   1525     @final
   1526     def __nonzero__(self) -> NoReturn:
-> 1527         raise ValueError(
   1528             f"The truth value of a {type(self).__name__} is ambiguous. "
   1529             "Use a.empty, a.bool(), a.item(), a.any() or a.all()."_

ValueError: The truth value of a Series is ambiguous. Use a.empty, a.bool(), a.item(), a.any() or a.all().